### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by ns-dell**
**Change log description**  
- Separates existing code for mirroring stream into committing metadata changes, actions and jobs.
- Pulling in kafka to pravega stream job

**Purpose of the change**  
- Pravega data importer

**What the code does**  
- Separates existing code for mirroring stream into committing metadata changes, actions and jobs.
- Pulling in kafka to pravega stream flink job, adding corresponding action

**How to verify it**  
- Run following commands in the directory:

Pravega to Pravega stream mirroring:
```
pravega-data-importer
  stream-mirroring
  input-controller=tcp://localhost:9090
  input-stream=source/sourceStream
  input-startAtTail=false
  output-stream=destination/destinationStream
  output-controller=tcp://127.0.0.1:9990
  flinkHost=localhost
  flinkPort=8081
```
Kafka to Pravega stream mirroring:
```
pravega-data-importer
  kafka-stream-mirroring
  input-topic=test-input
  bootstrap-servers=localhost:9092
  output-stream=destination/from-kafka
  output-controller=tcp://127.0.0.1:9090
  isStreamOrdered=true
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,106 @@
...
+flink run build/libs/pravega-data-importer-1.0-SNAPSHOT.jar \
+  --action-type stream-mirroring \
+  --input-controller tcp://localhost:9090 \
+  --input-stream examples/network \
+  --input-startAtTail false \
+  --output-stream examples2/network-cloud \
+  --output-controller tcp://127.0.0.1:9990
+```
+
+## Kafka-Stream-Mirroring: Continuously copying a Pravega stream to another Pravega stream
```
Comment (ShwethaSNayak) [inline README.md]: This should be Continuously copying a Kafka stream to Pravega stream

File: build.gradle [hunk 2]
```diff
@@ -39,31 +50,78 @@ repositories {
...
+    testAnnotationProcessor 'org.projectlombok:lombok:' + "${lombokVersion}"
+    implementation "org.apache.flink:flink-java:${flinkVersion}"
+    implementation "org.apache.flink:flink-streaming-java:${flinkVersion}"
+    implementation "org.apache.flink:flink-connector-kafka:${flinkVersion}"
+    implementation "org.apache.flink:flink-table-api-java-bridge:${flinkVersion}"
+    implementation "org.apache.flink:flink-table-planner_${flinkScalaVersion}:${flinkVersion}"
+    implementation "org.apache.flink:flink-clients:${flinkVersion}"
+    implementation "org.apache.flink:flink-runtime-web:${flinkVersion}"
+    implementation "org.apache.flink:flink-core:${flinkVersion}"
+//    implementation "io.pravega:pravega-connectors-flink-${flinkShortVersion}_${flinkScalaVersion}:${pravegaFlinkConnectorVersion}"
```
Comment (RaulGracia) [inline build.gradle]: Please, clean up from this file all the unnecessary imports and comments like this.

File: gradle.properties [hunk 3]
```diff
@@ -15,23 +15,32 @@ awsSDKVersion=1.11.762
 bucket4jVersion=4.4.1
```
Comment (RaulGracia) [inline gradle.properties]: Do we use all the variables defined here? If not, please delete the unused ones.

File: src/main/java/io/pravega/dataimporter/actions/AbstractAction.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+     * Returns the name of the job.
+     */
+    public abstract String getJobName();
+
+    /**
+     * Submits Flink job to Flink cluster, and logs the {@link org.apache.flink.api.common.JobID} of the submitted job.
+     */
+    public JobClient submitDataImportJob() {
+        JobClient jobClient = this.job.submitJob();
+        log.info("\n\n\nJob ID: " + jobClient.getJobID().toString() + "\n\n");
```
Comment (RaulGracia) [inline src/main/java/io/pravega/dataimporter/actions/AbstractAction.java]: Please, delete the "\n" from this log, it should be enough just printing the job id plus some test that describes what it is.

Comment (RaulGracia) [reply src/main/java/io/pravega/dataimporter/actions/AbstractAction.java]: Also, the signature of this method has changed, so the comments should change too. Why are you not using the regular Javadoc tags like `@param` or `@returns`?

File: src/main/java/io/pravega/dataimporter/jobs/AbstractJob.java [hunk 5]
```diff
@@ -17,58 +17,53 @@
...
+     * Used to retrieve the application configuration, containing parameters.
+     */
     public AppConfiguration getConfig() {
         return config;
     }
 
     /**
-     * If the Pravega stream does not exist, creates a new stream with the specified stream configuration.
-     * If the stream exists, it is unchanged.
+     * Concrete implementations use this method to submit Flink jobs to the Flink cluster.
```
Comment (RaulGracia) [inline src/main/java/io/pravega/dataimporter/jobs/AbstractJob.java:59]: Same in this file (and others) about javadoc.

**Review by RaulGracia — CHANGES_REQUESTED** [review]

**Review by RaulGracia — APPROVED** [review]

**Pull request merged**